src/MSAAdvanced.sol

```diff
@@ -195,6 +197,8 @@ contract MSAAdvanced is IMSA, ExecutionHelper, ModuleManager, HookManager {
     {
         (address hook, bytes memory hookData) = _preCheck();
 
+        if (!IModule(module).isModuleType(moduleTypeId)) revert MismatchModuleTypeId(moduleTypeId);
```

Comment:
imo this is not needed since this check is already done on installation right?

Reply:
Yes, we can remove it.

Reply:
wdym? my point is that we already check moduletype onInstallation, but I dont think theres a need to check on uninstallation since it doesnt rly matter anyways (bc the module wont be used anymore)

Reply:
should be good imo, on uninstall if module address is wrong or if it doesn't match typeid it would revert with sentinel list or in module itself. will share if anything post writing more tests